README.md

```diff
@@ -0,0 +1,219 @@
+```
+
+...but she discovered that the recipe was missing something. :confused:
+After a long searching process and many attempts she found the right missing ingredient.
+So Alice fixed the recipe and she replaced the old one with this new improved one. :pencil:
+
+```Typescript
+// Alice creates a recipe and replace the latest added with it
+    await cookBookInstance
+        .connect(ALICE)
+        .addAssetEntry(     // add a new asset to the collection (not binded to any token)
+            `ipsf://metadata/alice_recipe_${actual_recipe_id}`    // metadata URI of the asset
+        );
+    const recipe_TO_REPLACE = 6;
+    await cookBookInstance
+        .connect(ALICE)
+        .addAssetToToken(
+            ALICE_BOOK_ID,      // ID of the token that will receive the asset
+            actual_recipe_id,  // ID of the asset to add
+            recipe_TO_REPLACE  // ID of the asset to replace with the new one
+        );
+    actual_recipe_id++;    // Update to the next ID
+```
+
+After this fix alice's career took flight and Master, proud of his student, proposed her to create 2 recipes as a collaboration to add her cookbook. 
+
+```Typescript
+// Master creates 2 recipes and add them to Alice cookbook
+    for (let i = 0; i < 2; i++) {
+        await cookBookInstance
+            .connect(MASTER)
+            .addAssetEntry(     // add a new asset to the collection (not binded to any token)
+                `ipsf://metadata/master_recipe_${actual_recipe_id}`   // metadata URI of the asset
+            );
+        await cookBookInstance.addAssetToToken(
+            ALICE_BOOK_ID,      // ID of the token that will receive the asset
+            actual_recipe_id,  // ID of the asset to add
+            0                   // ID of the asset to replace with the new one. 0 == none
+        );
+        actual_recipe_id++;
+    }
+```
+
+After a small review Alice added them to her book. :book:
+
+```Typescript
+// Alice accepts recipes
+    for (let i = 0; i < 2; i++) {
+        const CURRENT_recipe_ID = actual_recipe_id - 2 + i;
+        let acceptTx = await cookBookInstance
+            .connect(ALICE)
+            .acceptAsset(           // accept pending asset
+                ALICE_BOOK_ID,      // ID of the token that will receive the asset
+                0,                  // Index of the asset in the pending array
+                CURRENT_recipe_ID  // ID of the asset to add
+            );
+        acceptTx.wait();
+    }
+```
+
+## Master retires and burn his cookbook - Token burning with related assets
+
+After this final collaboration with Alice Master decided to retire, he was an old man and he was also a little tired of travelling the world. This final decision took him to burn his cookbook. :fire:
+He wanted to leave only a nice memory of him, and a veil of mystery around his cooking figure :wave:
+
+```Typescript
+await cookBookInstance.connect(MASTER).
+        burn(MASTER_BOOK_ID // ID of the token to burn
+        );
+```
```

Comment:
We could clarify here that since the recipes (assets) are not tied to any token, new chefs could include them in their cookbooks, as long as they have the permission.
My point would makes more sense if only the contract owner (master) and contributors (alice) could add assets and add them to tokens. However, by looking at the script I see that alice was never added as a contributor, si she should be supposed to be adding assets nor adding them to tokens. How is it possible? :confused: